examples/RNOneSignal/App.js

```diff
@@ -8,484 +8,692 @@
-                onPress={() => {
-                  OneSignal.getTags(tags => {
-                    console.log('Did get tags: ', tags);
-
-                    this.setState({
-                      jsonDebugText: JSON.stringify(tags, null, 2),
-                    });
-                  });
-                }}
-                title="Get Tags"
-                color={this.state.buttonColor}
-              />
+
+    constructor(properties) {
+        super(properties);
+
+        this.state = {
+            // OneSignal states
+            // Privacy Consent states
+            hasPrivacyConsent: false, // App starts without privacy consent
+            isPrivacyConsentLoading: requiresPrivacyConsent,
+
+            // Device states
+            userId: '',
+            pushToken: '',
+
+            // Subscription states
+            isSubscribed: false,
+            isSubscriptionLoading: false,
+
+            // External User Id states
+            externalUserId: '',
+            isExternalUserIdLoading: false,
+
+            // Email states
+            email: '',
+            isEmailLoading: false,
+
+            // In-App Messaging states
+            iam_paused: false,
+
+            // Add more states here...
+
+            // Demo App states
+            debugText: ''
+        };
+
+        // Log level logcat is 6 (VERBOSE) and log level alert is 0 (NONE)
+        OneSignal.setLogLevel(6, 0);
+
+        // Share location of device
+        OneSignal.setLocationShared(true);
+
+        OneSignal.setRequiresUserPrivacyConsent(requiresPrivacyConsent);
+        OneSignal.init(appId, {
+            kOSSettingsKeyAutoPrompt: true,
+        });
+
+        // Notifications will display as NOTIFICATION type
+        OneSignal.inFocusDisplaying(2);
+
+        // If the app requires privacy consent check if it has been set yet
+        if (requiresPrivacyConsent) {
+        // async 'then' is only so I can sleep using the Promise helper method
+            OneSignal.userProvidedPrivacyConsent().then(async (granted) => {
+                // For UI testing purposes wait X seconds to see the loading state
+                await sleep(0);
+
+                console.log('Privacy Consent status: ' + granted);
+                this.setState({hasPrivacyConsent:granted, isPrivacyConsentLoading:false});
+            });
+        }
+
+        OneSignal.getPermissionSubscriptionState((response) => {
+            console.log('Device state:');
+            console.log(response);
+
+            let notificationsEnabled = response['notificationsEnabled'];
+            let isSubscribed = response['subscriptionEnabled'];
+
+            this.setState({isSubscribed:notificationsEnabled && isSubscribed, isSubscriptionLoading:false}, () => {
+                OneSignal.setSubscription(isSubscribed);
+            });
+        });
+
+        // Examples for using native IAM public methods
+//        this.oneSignalInAppMessagingExamples();
```

Comment:
can we remove commented code?

Reply:
no, it shows in our demo app we how to use some examples, I just comment them out to prevent unnecessary logs or confusion